packages/frontend/src/pages/Send/Send.tsx

```diff
@@ -287,14 +287,34 @@ const Send: FC = () => {
   const needsTokenForFee = useNeedsTokenForFee(fromNetwork)
 
   useEffect(() => {
-    if (needsTokenForFee && fromNetwork) {
-      setNeedsNativeTokenWarning(
-        `Add ${fromNetwork.nativeTokenSymbol} to your account on ${fromNetwork.name} for the transaction fee.`
-      )
-    } else {
-      setNeedsNativeTokenWarning('')
+    const update = async () => {
```

Comment:
since this hook makes async calls, we want to avoid any timing issues if this component gets unmounted for any reason. simple fix: follow the `isSubscribed` pattern that is [implemented here](https://github.com/hop-protocol/hop/blob/develop/packages/frontend/src/pages/Send/Send.tsx#L172)

[more info on cancelling hook promises](https://juliangaramendy.dev/blog/use-promise-subscription)

thanks, @miguelmota for the insight